### ISSUE TITLE
[Info] Current state of the project

### ISSUE THREAD

**Issue by fmorgner** [umurmur/umurmur#74]
In hope to reach everyone (at least the contributors that accepted the invitation to the organization way back when) I am opening this ticket to give you some status updates regarding the project.
# Website

Maybe some of you noticed that we have got a website by now. It is a very basic website thats currently hosted on GitHub Pages an can be reached at http://umurmur.net. If anyone would have the time and energy to improve on that it would be greatly appreciated.
# Communication

Even though there is currently a mailing list on [Google Groups](https://groups.google.com/forum/#!forum/umurmur-general) I am currently thinking about moving it on its own server and making it reachable at something like list@umurmur.net or the like. As always I am open to suggestions and feedback on this idea! I feel that having the list at least reachable via [something]@umurmur.net make a bit of a more "serious" impression.
# Repo organization

@snowblind wrote a new version of his monitoring tool an I incorporated it into the organization today. I will be writing a guide on how to contribute utilities using the shared memory API in the next couple of weeks.

Additionally I would like to suggest a little bit of a restructuring of the repo, namely moving the headers out of the 'src' directory an into their own. Please share your thoughts on that!
# Build systems

Speaking of reorganizing the repo I would like to ask if there is still a need for the somewhat old Autotools build environment. Personally i much prefer CMake since I feel it provides a lot more flexibility and a much more modern approach. BUT: If we really do need the Autotools environment for some good reason it will of course stay.
# Thanks

I would like to say thank you to every one of you for your contributions to the project, be it code, discussions, advice, reviews, documentation, you name it. At the same time I would like to say sorry for sometimes being unresponsive. This is my first time doing "project management" outside of school and I would like to assure you that I am trying to get better at it.

**Comment by Diaoul** [comment]
Moving to CMake is a good idea but keep in mind that it'll break all current packaging scripts. I'd say maintain both for a defined period of time with proper deprecation warning and then drop autotools.

**Comment by fmorgner** [comment]
Thats a very good point. Since I do not have any experience with packaging or reaching out to the package maintainers I am not sure on how to get in touch with them. Maybe I could try to reach them via the respective distribution repositories. I will keep that in mind. Thanks!

**Comment by snowblind** [comment]
You can move the SHM API headers to a new directory and the API docs could go in there too. It would be nice if we could make cmake find them automatically. It doesn't do it right for me here currently. I have to set  UMURMUR_ROOT_PATH on the cmake cmdline.

https://github.com/umurmur/numurmon
   The README.md file is using the same picture for both. Not showing the green for talking.

Neat way to hide the sreenshots dir and still be able to use them in the README.md had me confused for a minute.

**Comment by fmorgner** [comment]
@snowblind I fixed the image source, thanks for catching that. I think I also fixed the source location detection in both numurmon and umurmur-monitor. You should be able to work with it in two ways:

1. from within the tools/numurmon or tools/umurmur-monitor subdirectories
2. with seperate clones which are side-by-side to umurmur in the directory hierarchy

Please let me know if there are any more problems with CMake.

**Comment by snowblind** [comment]
@fmorgner I have a pull request up for the main umurmur repo

I will test the new cmake stuff soon. I still using my own git version of numurmon for now it works.
I worked on adding the missing server info and some code clean up.
I still need to write a test program to gen a programmable SHM area for testing handling users over 10, etc.

**Comment by snowblind** [comment]
@fmorgner  I am no cmake wiz but did you forget to include this in the cmake/Modules SelectRTBackend

It not setting the RT lib for linking.

**Comment by fmorgner** [comment]
I believe to have fixed that now. I missed a call to SelectRTBackend in both umurmur/numurmon and umurmur/umurmur-monitor. Could you verify if it works?

**Comment by snowblind** [comment]
@fmorgner It works now. Do you know why it makes me type user name and password for ssh access to umumur org gits. Do you need to add my pub key to the umumur org? Can you even do that?

I really need to change my password for github and when I do it is not going to be easily type able

**Comment by fmorgner** [comment]
@snowblind Sounds like it is trying to push vi https. Have you changed the url of the 'origin' remote to the  ssh URL (git@github.com:umurmur/numumon.git)? Or have you added an upstream remote? In the latter case you would need to specify that remote when pushing (git push upstream master). What is the output of `git remote -v` when executed in the numurmon subdirectory?

**Comment by snowblind** [comment]
Yes that was what was wrong I had https in remote urls. I need to not wait for months between uses of git I forget stuff I already learned once.

**Comment by hasufell** [comment]
> Moving to CMake is a good idea but keep in mind that it'll break all current packaging scripts. I'd say maintain both for a defined period of time with proper deprecation warning and then drop autotools.

As a packager I very much prefer the hard way. Always have only one build system. There are a few upstreams that maintain multiple ones and it has always caused trouble, because the build systems end up somewhat out of sync.

Packagers will notice the change, because that's what they do.

About cmake: I don't really agree that it's a better choice. Just a few points that I dislike about it:
* cmake (when used outside of a package manager) doesn't provide _any_ uninstall rules and that's a "design decision" by cmake upstream... some people provide their own uninstall rules, which can often go wrong (it did go wrong in blender for example)
* cmakes way to look for libraries relies on a lot of 3rd party cmake modules, that sometimes do really weird stuff and are not maintained by cmake developers
* although cmake has support for pkgconfig, using it is somewhat inconvenient and can be hard to get right

The last two points often require distributions to heavily patch modules and build systems to make stuff compile.

The only downsides of autotools are the complexity and the bootstrapping. The latter is a non-issue for releases, because it will already have a bootstrapped configure file.
